exporter/awsemfexporter/metric_translator.go

```diff
@@ -118,6 +134,268 @@ func (dps DoubleHistogramDataPointSlice) At(i int) DataPoint {
+			ilm := ilms.At(j)
+			if ilm.IsNil() {
+				continue
+			}
+
+			if ilm.InstrumentationLibrary().IsNil() {
+				instrumentationLibName = noInstrumentationLibraryName
+			} else {
+				instrumentationLibName = ilm.InstrumentationLibrary().Name()
+			}
+
+			metrics := ilm.Metrics()
+			for k := 0; k < metrics.Len(); k++ {
+				metric := metrics.At(k)
+				if metric.IsNil() {
+					totalDroppedMetrics++
+					continue
+				}
+
+				switch metric.DataType() {
+					case pdata.MetricDataTypeIntGauge:
+						dps = IntDataPointSlice{metric.IntGauge().DataPoints()}
+					case pdata.MetricDataTypeDoubleGauge:
+						dps = DoubleDataPointSlice{metric.DoubleGauge().DataPoints()}
+					case pdata.MetricDataTypeIntSum:
+						dps = IntDataPointSlice{metric.IntSum().DataPoints()}
+					case pdata.MetricDataTypeDoubleSum:
+						dps = DoubleDataPointSlice{metric.DoubleSum().DataPoints()}
+					case pdata.MetricDataTypeDoubleHistogram:
+						dps = DoubleHistogramDataPointSlice{metric.DoubleHistogram().DataPoints()}
+					default:
+						config.logger.Warn("Unhandled metric data type.",
+							zap.String("DataType", metric.DataType().String()),
+							zap.String("Name", metric.Name()),
+							zap.String("Unit", metric.Unit()),)
+						continue
+				}
+
+				if dps.Len() == 0 {
+					continue
+				}
+
+				for m := 0; m < dps.Len(); m++ {
+					dp := dps.At(m)
+					if dp.IsNil() {
+						continue
+					}
+
+					fields := make(map[string]interface{})
+					fields[nameSpace] = namespace
+					fields[OTellibDimensionKey] = instrumentationLibName
+	
+					dp.LabelsMap().ForEach(func(k string, v string) {
+						fields[k] = v
+					})
+					
+					s := make([]string, len(fields)*2)
+					for k, v := range fields {
+						if (k != OTellibDimensionKey) {
+							s = append(s, k, v.(string))
+						} else {
+							s = append(s, k, v.(string))
+						}
+					}
+					sort.Strings(s)
+					key := strings.Join(s, "")
+
+					if _, ok := groupedMetricMap[key]; ok {
+						updateGroupedMetric(dp, &metric, key, groupedMetricMap)
+					} else {
+						groupedMetric := buildGroupedMetric(dp, fields, &metric, config.DimensionRollupOption)
+						if groupedMetric != nil {
+							groupedMetricMap[key] = groupedMetric
+						}
+					}
+				}
+			}
+		}
+	}
+	return groupedMetricMap, totalDroppedMetrics
+}
+
+// Build grouped metric from Datapoint and pdata.Metric
+func buildGroupedMetric (dp DataPoint, fields map[string]interface{}, pMetricData *pdata.Metric, dimensionRollupOption string) (*GroupedMetric) {
+	var namespace string
+	timestamp := time.Now().UnixNano() / int64(time.Millisecond)
+	labels := make(map[string]string)
+	metrics := make(map[string]MetricInfo)
+	
+	namespace = fields[nameSpace].(string)
+
+	// Extract labels
+	for k, v := range fields {
+		if k != nameSpace {
+			labels[k] = v.(string)
+		}
+	}
+
+	// Extract metric
+	var metricVal interface{}
+	switch metric := dp.(type) {
+	case pdata.IntDataPoint:
+		metricVal = int64(metric.Value())
```

Comment:
I think you're missing some logic here. Please run unit tests before submitting PR to ensure that all unit tests run correctly.